kyo-core/shared/src/main/scala/kyo/scheduler/IOTask.scala

```diff
@@ -55,7 +55,7 @@ private[kyo] class IOTask[Ctx, E, A] private (
                                 val trace = this.trace
                                 this.trace = null.asInstanceOf[Trace]
                                 input.onComplete { r =>
-                                    val task = IOTask(IO(cont(r.asInstanceOf[Result[Nothing, C]])), trace, context, ensures, runtime)
```

Comment:
The `IO` suspension here isn't strictly necessary since this is already unsafe execution. It might execute more pure transformations during flushing but it should be ok. For reference, Twitter's `Future` always executes continuations in the thread that flushes the continuations.

Reply:
I'm reverting this change. There's actually a good reason for the `IO` suspension here: error handling. If the continuation throws when applied, the failure won't be piped correctly without `IO`.